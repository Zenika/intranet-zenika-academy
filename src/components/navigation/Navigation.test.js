import React from 'react';
import ReactDOM from 'react-dom';
import TestUtils from 'react-dom/test-utils';
import { shallow } from 'enzyme';
import NavigationBar from './Navigation';

describe('Navigation bar when the user isn\'t connected', () => {
  const nav = <NavigationBar />;
  test('Navigation bar exists', () => {
    expect(nav).toBeDefined();
    expect(TestUtils.isElement(nav)).toBe(true);
    expect(TestUtils.isElementOfType(nav, NavigationBar)).toBe(true);

    const rendered = TestUtils.renderIntoDocument(<NavigationBar />);
    const siButton = TestUtils.findRenderedDOMComponentWithClass(rendered, 'signInDesktop');

    expect(siButton.textContent).toEqual('Se connecter');
  });

  test('Shallow rendering', () => {
    const navbar = shallow(<NavigationBar />);
    const navbarInstance = navbar.instance();

    navbarInstance.componentDidMount();

    expect(navbar.state('modalState')).toBe('');
  });
});
