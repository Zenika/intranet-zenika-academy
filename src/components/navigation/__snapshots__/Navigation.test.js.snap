// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly Displays the connexion Modal when sign in button is clicked 1`] = `
<NavigationBar>
  <nav
    aria-label="main navigation"
    className="navbar is-primary"
    role="navigation"
  >
    <section
      className="navbar-item display-mobile"
    >
      <button
        className="button"
        onClick={[Function]}
        type="submit"
      >
        Se connecter
      </button>
    </section>
    <section
      className="navbar-menu"
    >
      <section
        className="navBarLoggedOut"
      >
        <section
          className="navbar-start"
        >
          <p>
            There is no fate but what we make
          </p>
        </section>
      </section>
      <section
        className="navbar-end"
      >
        <section
          className="navbar-item"
        >
          <button
            className="button signInDesktop"
            onClick={[Function]}
            type="submit"
          >
            Se connecter
          </button>
        </section>
      </section>
    </section>
    <SignInModal
      connect={[Function]}
      toggleModal={[Function]}
    >
      <section
        className="modal is-active"
        id="modalLogin"
      >
        <section
          className="modal-background"
        />
        <section
          className="modal-card"
        >
          <header
            className="modal-card-head"
          >
            <p
              className="modal-card-title"
            >
              Connexion
            </p>
          </header>
          <section
            className="modal-card-body"
          >
            <form>
              <section
                className="field"
              >
                <p
                  className="control has-icons-left"
                >
                  <input
                    className="input"
                    id="emailInput"
                    name="email"
                    onChange={[Function]}
                    placeholder="Email"
                    required={true}
                    type="text"
                    value=""
                  />
                  <span
                    className="icon is-small is-left"
                  />
                </p>
              </section>
              <section
                className="field"
              >
                <p
                  className="control has-icons-left"
                >
                  <input
                    className="input"
                    id="passwordInput"
                    name="password"
                    onChange={[Function]}
                    placeholder="Password"
                    required={true}
                    type="password"
                    value=""
                  />
                  <span
                    className="icon is-small is-left"
                  >
                    <i
                      className="fas fa-lock"
                    />
                  </span>
                </p>
              </section>
            </form>
          </section>
          <footer
            className="modal-card-foot modalFooter"
          >
            <button
              className="button is-success"
              id="submitButton"
              onClick={[Function]}
              type="submit"
            >
              Se connecter
            </button>
            <button
              className="button"
              id="cancelButton"
              onClick={[Function]}
              type="button"
            >
              Annuler
            </button>
          </footer>
        </section>
      </section>
    </SignInModal>
  </nav>
</NavigationBar>
`;

exports[`Navbar renders correctly it renders correctly 1`] = `
<NavigationBar>
  <nav
    aria-label="main navigation"
    className="navbar is-primary"
    role="navigation"
  >
    <section
      className="navbar-item display-mobile"
    >
      <button
        className="button"
        onClick={[Function]}
        type="submit"
      >
        Se connecter
      </button>
    </section>
    <section
      className="navbar-menu"
    >
      <section
        className="navBarLoggedOut"
      >
        <section
          className="navbar-start"
        >
          <p>
            There is no fate but what we make
          </p>
        </section>
      </section>
      <section
        className="navbar-end"
      >
        <section
          className="navbar-item"
        >
          <button
            className="button signInDesktop"
            onClick={[Function]}
            type="submit"
          >
            Se connecter
          </button>
        </section>
      </section>
    </section>
  </nav>
</NavigationBar>
`;
